pontoon/base/models.py

```diff
@@ -540,27 +536,53 @@ def get_entities(project, locale, paths=None):
         resources = resource_with_paths or resources
 
     entities = Entity.objects.filter(resource__in=resources, obsolete=False)
-    entities_array = []
 
-    for e in entities:
+    entities_array = entities.values(
+        'pk', 'string', 'string_plural', 'key', 'resource__path',
+        'resource__format', 'comment', 'order', 'source', 'obsolete'
+    )
+
+    translations_array = Translation.objects.filter(locale=locale, entity__in=entities).values()
+    translations_dict = {}
+
+    for t in translations_array:
+        translations_dict.setdefault(t['entity_id'], []).append(t)
```

Comment:
There's a class in the stdlib called [`collections.defaultdict`](https://docs.python.org/2/library/collections.html#collections.defaultdict) that provides this functionality for you.